src/test/skript/tests/general/EquippableComponents.sk

```diff
@@ -0,0 +1,206 @@
+	set the camera overlay of {_component} to {@camera}
+	assert the camera overlay of {_component} is {@camera} with "Camera overlay of component was not set"
+	allow {_component} to swap equipment
+	assert {_component} can swap equipment with "Component should be swappable"
+	prevent {_component} from swapping equipment
+	assert {_component} can not swap equipment with "Component should not be swappable"
+	allow {_component} to be dispensed
+	assert {_component} can be dispensed with "Component should be dispensable"
+	prevent {_component} from being dispensed
+	assert {_component} can not be dispensed with "Component should not be dispensable"
+	make {_component} lose durability when hurt
+	assert {_component} will lose durability when hurt with "Component should lose durability when hurt"
+	make {_component} not lose durability when hurt
+	assert {_component} will not lose durability when hurt with "Component should not lose durability when hurt"
+
+	parse if method {@method} exists:
+		allow {_component} to be equipped onto entities
+		assert {_component} can be equipped onto entities with "Component should equip onto entities"
+		prevent {_component} from being equipped onto entities
+		assert {_component} can not be equipped onto entities with "Component should not equip onto entities"
+	set {_original} to {_component}
+
+	set {_copy} to a item component copy of {_original}
+	set the equip sound of {_copy} to "item.armor.equip_diamond"
+	assert the equip sound of {_copy} is "item.armor.equip_diamond" with "Equip sound of copy was not set"
+	assert the equip sound of {_copy} is not the equip sound of {_original} with "Equip sound of copy should not be the same as original"
+	set the equipment slot of {_copy} to helmet slot
+	assert the equipment slot of {_copy} is helmet slot with "Equipment slot of copy was not set"
+	assert the equipment slot of {_copy} is not the equipment slot of {_original} with "Equipment slot of copy should not be the same as original"
+	set the equipped model key of {_copy} to "skript:model_key_copy"
+	assert the equipped model key of {_copy} is "skript:model_key_copy" with "Model of copy was not set"
+	assert the equipped model key of {_copy} is not the equipped model key of {_original} with "Model key of copy should not be the same as original"
+	set the allowed entities of {_copy} to a villager and a zombie villager
+	assert the allowed entities of {_copy} is a villager and a zombie villager with "Allowed entities of copy was not set"
+	assert the allowed entities of {_copy} is not the allowed entities of {_original} with "Allowed entities of copy should not be the same as original"
+	set the camera overlay of {_copy} to "skript:camera_key_copy"
+	assert the camera overlay of {_copy} is "skript:camera_key_copy" with "Camera overlay of copy was not set"
+	assert the camera overlay of {_copy} is not the camera overlay of {_original} with "Camera overlay of copy should not be the same as original"
```

Comment:
assertions on the original values still needs to be addressed, as mentioned on discord.